packages/core/src/config/promotion/actions/order-fixed-discount-action.ts

```diff
@@ -13,7 +13,7 @@ export const orderFixedDiscount = new PromotionOrderAction({
         },
     },
     execute(ctx, order, args) {
-        return -args.discount;
+        return -Math.min(args.discount, order.total);
```

Comment:
ah you use the `order.total` rather than `totalWithTax` because the promotions all operate pre-tax, right? Otherwise we could still end up negative. Is that the case?